### DISCUSSION TITLE
Built-in Thumbnail Generator

### DISCUSSION THREAD

**Discussion by CoreyPenIT** [smplayer-dev/smplayer#695] [General]
I'm just curious if anyone can shed some light on the method smplayer uses to generate thumbnails.  Is it custom code, ffmpeg, or a combination of things?  I'd like to have this functionality from cli where I can run it for multiple files etc without having to use a gui.  I've tried different tools but none achieve exactly the results smplayer does.

**Comment by smplayer-dev** [comment]
It calls mplayer or mpv to extract the pictures. Take a look at the [videopreview](https://github.com/smplayer-dev/smplayer/tree/master/src/videopreview) folder. It can even be built as a standalone application.
